src/utils/safeBatchGetItem.js

```diff
@@ -0,0 +1,70 @@
+const _ = require('lodash');
+
+const concatIfArray = (objValue, srcValue) => {
+  if (_.isArray(objValue) && _.isArray(srcValue)) {
+    return [...objValue, ...srcValue];
+  }
+
+  return undefined;
+};
+
+// DO NOT MODIFY, this value is specified in dynamodb docs
+const maxKeys = 100;
+
+/**
+ * A wrapper for dynamodb's batchGetItem
+ * This wrapper handles requests of more than 100 items at the same time
+ *  by splitting them into separate batchGetItems
+ *
+ * @param {*} dynamodb dynamodb sdk client
+ * @param {*} params params as would be passed to batchGetItem
+ * @returns The result as would be returned by batchGetItem
+ */
+const safeBatchGetItem = async (dynamodb, params) => {
+  const tableNames = [];
+  const chunkedKeysByTableName = {};
+
+  let amountOfRequests = 0;
+
+  Object.entries(params.RequestItems).forEach(([tableName, { Keys: keys }]) => {
+    tableNames.push(tableName);
+    chunkedKeysByTableName[tableName] = _.chunk(keys, maxKeys);
+
+    amountOfRequests = Math.max(chunkedKeysByTableName[tableName].length, amountOfRequests);
+  });
+
+  const requestPromises = _.range(amountOfRequests).map(async (index) => {
+    const currentKeys = {};
+
+    tableNames.forEach((tableName) => {
+      const keysForTable = chunkedKeysByTableName[tableName][index];
```

Comment:
So the keys for the table would be in a different increasing index every time? How is the structure of this?

Reply:
I use _.chunk() with partitions of size 100, which works like this example:
If I have a list `a = [0,1,2,3,4,5,6,7]` 
then from doing `_.chunk(a, 4)` I get `[ [0,1,2,3], [4,5,6,7] ]`

Reply:
Yeah, `partitionIndex` sounds good

Reply:
Ok, did that change andadded some comments